### ISSUE TITLE
carkeypoints in pytorch and the content of the result.txt

### ISSUE THREAD

**Issue by shen2106** [krrish94/CarKeypoints#5]
Hello! Thanks for your great work on this subject.
Will you be so glad to answer a few questions of mine which are listed below?
1.Is there any available tensorflow or pytorch version of it?If yes, how can I get the code and the pre-trained weight to detect keypoints of cars? If not, then I want to build a  network similar to Stackedhourglass to extract these 36 keypoints of cars.Which dataset did you use when you trained this Carkeypoints network?
2.I know that the content of the result.txt is made up of three parts and two of them are the x coordinate and the y coordinate of the keypoints.I am confused about the detailed meaning of the third part and the range of it.Would you please explain it for me.
Thank you for your help so much!

**Comment by krrish94** [comment]
Hi @shen2106 

1. Unfortunately not. We only have the Lua torch version of this network. We trained this on a synthetic dataset comprising CAD models of cars from ShapeNet which we rendered to 2D. We also finetuned on a handful of real samples, but the model trained on synthetic data alone works fine enough on KITTI. If you're trying to train a pytorch/tensorflow version of this, I'd recommend starting out with an existing stacked hourglass repo, and retraining it on KITTI data  ( http://www.zeeshanzia.com/dataset/KITTI3D_cvpr17.zip or http://www.nec-labs.com/~mas/DISCO/ ).

2. Yes, the first two columns denote the x and y coordinates of each keypoint. The third column denotes the softmax scores at that particular x, y location. This "weakly" indicates confidence of that particular keypoint. (Note that, of late, it has been shown that such softmax confidence scores cannot be used as reliable measures of accuracy).
